### ISSUE TITLE
PDB parser does not recover all types from a PDB

### ISSUE THREAD

**Issue by yrp604** [Vector35/binaryninja-api#2932]
**Version and Platform (required):**
- Binary Ninja Version: 3245-dev
- OS: Windows
- OS Version: 11

**Bug Description:**
Loading a PDB from ntos, the pdb contains structures which are not successfully imported

**Steps To Reproduce:**
Please provide all steps required to reproduce the behavior:
1. Download the attached exe and pdb
2. Load the exe and the pdb
3. In the types view, look for _KENLISTMENT. Currently I see:
![image](https://user-images.githubusercontent.com/8716677/152481658-8662d18e-eee4-4f43-8823-794146ebf907.png)

4. Download pdbex from https://github.com/wbenny/pdbex
5. Run pdbex on the pdb with `pdbex _KENLISTMENT ntkrnlmp.pdb`, you should see a bunch of text finishing in:
```
# snip...
typedef struct _KENLISTMENT
{
  /* 0x0000 */ unsigned long cookie;
  /* 0x0004 */ long Padding_0;
  /* 0x0008 */ struct _KTMOBJECT_NAMESPACE_LINK NamespaceLink;
  /* 0x0030 */ struct _GUID EnlistmentId;
  /* 0x0040 */ struct _KMUTANT Mutex;
  /* 0x0078 */ struct _LIST_ENTRY NextSameTx;
  /* 0x0088 */ struct _LIST_ENTRY NextSameRm;
  /* 0x0098 */ struct _KRESOURCEMANAGER* ResourceManager;
  /* 0x00a0 */ struct _KTRANSACTION* Transaction;
  /* 0x00a8 */ enum _KENLISTMENT_STATE State;
  /* 0x00ac */ unsigned long Flags;
  /* 0x00b0 */ unsigned long NotificationMask;
  /* 0x00b4 */ long Padding_1;
  /* 0x00b8 */ void* Key;
  /* 0x00c0 */ unsigned long KeyRefCount;
  /* 0x00c4 */ long Padding_2;
  /* 0x00c8 */ void* RecoveryInformation;
  /* 0x00d0 */ unsigned long RecoveryInformationLength;
  /* 0x00d4 */ long Padding_3;
  /* 0x00d8 */ void* DynamicNameInformation;
  /* 0x00e0 */ unsigned long DynamicNameInformationLength;
  /* 0x00e4 */ long Padding_4;
  /* 0x00e8 */ struct _KTMNOTIFICATION_PACKET* FinalNotification;
  /* 0x00f0 */ struct _KENLISTMENT* SupSubEnlistment;
  /* 0x00f8 */ void* SupSubEnlHandle;
  /* 0x0100 */ void* SubordinateTxHandle;
  /* 0x0108 */ struct _GUID CrmEnlistmentEnId;
  /* 0x0118 */ struct _GUID CrmEnlistmentTmId;
  /* 0x0128 */ struct _GUID CrmEnlistmentRmId;
  /* 0x0138 */ unsigned long NextHistory;
  /* 0x013c */ struct _KENLISTMENT_HISTORY History[20];
  /* 0x01dc */ long __PADDING__[1];
} KENLISTMENT, *PKENLISTMENT; /* size: 0x01e0 */
```

**Expected Behavior:**
I expect binary ninja to recover all types.

**Additional Information:**
[ntkrnlmp.pdb.txt](https://github.com/Vector35/binaryninja-api/files/8000556/ntkrnlmp.pdb.txt)
[ntoskrnl.txt](https://github.com/Vector35/binaryninja-api/files/8000557/ntoskrnl.txt)

**Comment by psifertex** [comment]
Thanks for the report -- this is a high priority fix for our 3.1 release.

**Comment by CouleeApps** [comment]
Resolved as of 3.1.3715-dev with the new PDB loader.
